assets/js/googlesitekit/datastore/user/feature-tours.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * `core/user` data store: feature tours
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { STORE_NAME } from './constants';
+import { baseControls } from '../../modules/datastore/modules';
+import { createRegistrySelector } from '@wordpress/data';
+
+// Actions.
+const DISMISS_TOUR = 'DISMISS_TOUR';
+
+const fetchGetDismissedToursStore = createFetchStore( {
+	baseName: 'getDismissedTours',
+	controlCallback: () => {
+		return API.get( 'core', 'user', 'dismissed-tours', {}, { useCache: false } );
+	},
+	reducerCallback: ( state, dismissedTours ) => {
+		return {
+			...state,
+			dismissedTours,
+		};
+	},
+} );
+const { fetchGetDismissedTours, receiveGetDismissedTours } = fetchGetDismissedToursStore.actions;
+const fetchDismissTourStore = createFetchStore( {
+	baseName: 'dismissTour',
+	controlCallback: ( { slug } ) => API.set( 'core', 'user', 'dismiss-tour', { slug } ),
+	reducerCallback: ( state, dismissedTours ) => {
+		return {
+			...state,
+			dismissedTours,
+		};
+	},
+	argsToParams: ( slug ) => ( { slug } ),
+	validateParams: ( { slug } = {} ) => {
+		invariant( slug, 'slug is required.' );
+	},
+} );
+const { fetchDismissTour } = fetchDismissTourStore.actions;
+
+const baseInitialState = {
+	dismissedTours: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Dismisses the given tour by slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} slug Tour slug to dismiss.
+	 * @return {Object} Generator instance.
+	 */
+	dismissTour( slug ) {
+		invariant( slug, 'A tour slug is required to dismiss a tour.' );
+
+		return ( function* () {
+			// Dismiss the given tour immediately.
+			yield {
+				payload: { slug },
+				type: DISMISS_TOUR,
+			};
+			// Dispatch a request to persist the dismissal.
+			const { response, error } = yield fetchDismissTour( slug );
+			if ( ! error ) {
+				yield receiveGetDismissedTours( response );
+			}
+			return { response, error };
+		}() );
```

Comment:
Why not make this a generator action like we have elsewhere for these? What is the benefit of this approach? From a code readability perspective this seems unnecessarily complicated.

Reply:
@felixarntz It's needed for validation to work as expected. If you recall, we do the same thing in `createFetchStore` this is just the inline version of it:
https://github.com/google/site-kit-wp/blob/fb2bbdf94d435753a3c84f2a1064108144e390df/assets/js/googlesitekit/data/create-fetch-store.js#L176-L185

Reply:
Ah right, makes sense.